src/protocols/user_directed_commissioning/UserDirectedCommissioningClient.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements an object for a CHIP Echo unsolicitied
+ *      initiator (client).
+ *
+ */
+
+#include "UserDirectedCommissioning.h"
+
+namespace chip {
+namespace Protocols {
+namespace UserDirectedCommissioning {
+
+CHIP_ERROR UserDirectedCommissioningClient::SendUDCMessage(TransportMgrBase * transportMgr, System::PacketBufferHandle && payload,
```

Comment:
This does not seem to set up things like the exchange id correctly...  It will always use 0.  Might not matter in practice because no response is expected, but spec does not say to set the exchange id to 0 here.  

It might have been better if this shared more of the normal send path instead of copy/pasting bits of it and creating a parallel but somewhat weird setup.

Reply:
What is the exchange id in a message that does not have a response? The spec does currently provide guidance as far as I can tell.

Reply:
Per spec, there should be a new exchange id for every message that is not part of an existing exchange....